### PR TITLE
Sort exposures in unit test to make the result deterministic

### PR DESCRIPTION
**Pull request by chaodaiG**
This test failed randomly with out of order problem https://oss-prow.knative.dev/view/gs/oss-prow/pr-logs/pull/google_exposure-notifications-server/981/pull-en-server-release-unit/1304300664213475328

/assign @sethvargo

### CODE REVIEW

File: internal/publish/database/exposure_test.go [hunk 1]
```diff
@@ -756,6 +756,10 @@ func TestIterateExposuresCursor(t *testing.T) {
 	if err == nil {
 		t.Fatal("expected error")
 	}
+	// IterateExposures doesn't guarantee the order, sort to make this test deterministic
+	sort.Slice(seen, func(i, j int) bool {
+		return seen[i].IntervalNumber < seen[j].IntervalNumber
```
Comment (sethvargo) [inline internal/publish/database/exposure_test.go:761]: Do you want to sort by b64 TEK, then IntervalNumber, then rolling period?

Comment (chaodaiG) [reply internal/publish/database/exposure_test.go:761]: For this case I think IntervalNumber is sufficient. Also a QQ: how do we compare b64 TEK?

Comment (sethvargo) [reply internal/publish/database/exposure_test.go:761]: It's just a string, so `<` or `>`.

Comment (chaodaiG) [reply internal/publish/database/exposure_test.go:761]: I thought they are generated from random hashes, are they guaranteed to be ordered?

Comment (sethvargo) [reply internal/publish/database/exposure_test.go:761]: They are random, but when you're sorting two lists to compare them, you just sort them like strings.

Comment (sethvargo) [reply internal/publish/database/exposure_test.go:761]: Yea

File: internal/publish/database/exposure_test.go [hunk 2]
```diff
@@ -756,6 +756,10 @@ func TestIterateExposuresCursor(t *testing.T) {
 	if err == nil {
 		t.Fatal("expected error")
 	}
+	// IterateExposures doesn't guarantee the order, sort to make this test deterministic
```
Comment (mikehelmick) [inline internal/publish/database/exposure_test.go:759]: the preferred way to do this is to pass a sort function to cmp.Diff as a cmpopt

https://godoc.org/github.com/google/go-cmp/cmp/cmpopts#SortSlices

Comment (chaodaiG) [reply internal/publish/database/exposure_test.go:759]: Good to learn, done.

File: internal/publish/database/exposure_test.go [hunk 3]
```diff
@@ -756,7 +756,11 @@ func TestIterateExposuresCursor(t *testing.T) {
 	if err == nil {
 		t.Fatal("expected error")
 	}
-	if diff := cmp.Diff(exposures[:2], seen, ignoreUnexportedExposure); diff != "" {
+	// IterateExposures doesn't guarantee the order, sort to make this test deterministic
+	sortExposureFunc := func(i, j int) bool {
```
Comment (mikehelmick) [inline internal/publish/database/exposure_test.go:760]: I think this needs to take in two `*model.Exposure` params and return which true if the first is "less than" the other

Comment (chaodaiG) [reply internal/publish/database/exposure_test.go]: That's right, fixed.

Surprisingly the previous one worked...

**Review by mikehelmick — APPROVED** [review]

**Pull request merged**